lib/jsdom/living/nodes/HTMLAnchorElement-impl.js

```diff
@@ -9,10 +9,16 @@ class HTMLAnchorElementImpl extends HTMLElementImpl {
     super(globalObject, args, privateData);
 
     this._htmlHyperlinkElementUtilsSetup();
+  }
 
-    this._hasActivationBehavior = true;
+  // An anchor element has an activation behavior only if it creates a hyperlink.
+  // An anchor element represents a hyperlink if it has an href attribute.
+  get _hasActivationBehavior() {
+    return this.hasAttributeNS(null, "href");
   }
 
+  set _hasActivationBehavior(v) {}
```

Comment:
Is the setter necessary?

Reply:
Remove those two setters will cause error, becauce the parent class `HTMLElement` need set `_hasActivationBehavior`. 
https://github.com/zjffun/jsdom/blob/rm-assert-throw/lib/jsdom/living/nodes/HTMLElement-impl.js#L23

Reply:
Seems a bug.
```js
class C1 {
    constructor() {
        this.foo = 'test';
    }
}

class C2 extends C1 {
    get foo(){}
}

let obj = new C2
/**
Uncaught TypeError: Cannot set property foo of #<C2> which has only a getter
    at new C1 (repl:3:18)
    at new C2 (repl:1:1)
*/
```

```js
class C3 {
    get foo(){}
}

let obj2 = new C3
obj2.foo = 'test' // ok
```